SignalExtraction/fitWe.C

```diff
@@ -515,129 +533,171 @@ void fitWe(const TString  outputDir,   // output directory
           corrMet=met, corrMetPhi=metPhi;
         } else { 
-          recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-          hAntiWenuMetm->Fill(corrMet,weight2); //*w2);
+          hWenuMetm_RecoilDown->Fill(corrMet,weight); 
           corrMet=met, corrMetPhi=metPhi;
         }
-      }
-    }
-	  //         }
-	  //       if(typev[ifile]==eWenup1){
-	  //         Double_t corrMet=met, corrMetPhi=metPhi;
-	  //         Double_t lepPt = lep->Pt();
-	  //         Double_t lepPtup = lep->Pt();
-	  //         Double_t lepPtdown = lep->Pt();
-	  //         if(lepPtup        > PT_CUT) {
-	  //           corrMet=met, corrMetPhi=metPhi;
-	  //           hWenuMet_ScaleUp->Fill(corrMet,weight);
-	  //           if(q>0) {
-	  // //             recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-	  //             hWenuMetp_ScaleUp->Fill(corrMet,weight); 
-//             corrMet=met, corrMetPhi=metPhi;
-//           } else { 
-// //           recoilCorrm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-//           hWenuMetm_ScaleUp->Fill(corrMet,weight); 
-//           corrMet=met, corrMetPhi=metPhi;
-//           }
-// 	    }
-// //       }
-// //       if(typev[ifile]==eWenum1){
-// //         Double_t corrMet=met, corrMetPhi=metPhi;
-// //         Double_t lepPt = lep->Pt();
-// //         Double_t lepPtup = lep->Pt();
-// //         Double_t lepPtdown = lep->Pt();
-//         if(lepPtdown        > PT_CUT) {
-// 	      corrMet=met, corrMetPhi=metPhi;
-// 	      hWenuMet_ScaleDown->Fill(corrMet,weight);
-//           corrMet=met, corrMetPhi=metPhi;
-// 	      if(q>0) { 
-// //           recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-// 		  hWenuMetp_ScaleDown->Fill(corrMet,weight); 
-//           corrMet=met, corrMetPhi=metPhi;
-//           } else { 
-// //           recoilCorrm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
-// 		  hWenuMetm_ScaleDown->Fill(corrMet,weight); 
-//           corrMet=met, corrMetPhi=metPhi;
-//           }
-// 	    }
-// //       }
-        if(typev[ifile]==eEWK) {
+	  }
+	}
+	else if(typev[ifile]==eAntiWenu)
+	  {
+	    Double_t corrMet=met, corrMetPhi=metPhi;
+	    if(lepPt        > PT_CUT) {
+	      hAntiWenuMet->Fill(corrMet,weight2);
+	      if(q>0) {
+            recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+//             TVector2 vMetCorr((corrMet)*cos(corrMetPhi),(corrMet)*sin(corrMetPhi));
+//             Double_t corrMetWithLepton = (vMetCorr + vLepRaw - vLepCor).Mod();
+            Double_t corrMetWithLepton = met;
+            hAntiWenuMetp->Fill(corrMetWithLepton,weight2); // *w2 
+            corrMet=met, corrMetPhi=metPhi;
+	      } else { 
+            recoilCorr->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
```

Comment:
The recoil correction applied in the anti-isolation sample use the 0-0.5 bin to be applied to all the bins. need to include the inclusive bins since this is also used for the systematics